### ISSUE TITLE
tsgo does not report TS2565 for conditionally-assigned expando function properties

### ISSUE THREAD

**Issue by mds-ant** [microsoft/typescript-go#4134]
## Steps to reproduce

`main.ts`:
```ts
declare var b: boolean;
function d() {}
if (b) {
  d.q = false;
}
d.q;
```

```sh
tsgo main.ts
```

## Behavior with `typescript@6.0`

```text
main.ts(6,3): error TS2565: Property 'q' is used before being assigned.
```

## Behavior with `tsgo`

No error.
